src/Event/EuLoginEventSubscriber.php

```diff
@@ -68,6 +81,14 @@ public static function getSubscribedEvents(): array {
    */
   public function generateEmail(CasPreRegisterEvent $event): void {
     $attributes = $event->getCasPropertyBag()->getAttributes();
+    $user_settings = $this->configFactory->get('user.settings');
```

Comment:
This method has gone way past its respos. So let's either rename it to be more generic or break it up into multiple ones.

Reply:
Method was renamed to "processUserProperties", conversation can be resolved.